File: picker/src/main/java/com/braver/tool/picker/BraveFileGetter.java
package com.braver.tool.picker;

import android.content.Context;
import android.graphics.Bitmap;
import android.net.Uri;

import java.io.File;

public class BraveFileGetter {

    public static String getFileType(BraveFileType braveFileType) {
        switch (braveFileType) {
            case IMAGE:
                return "image";
            case VIDEO:
                return "video";
            case DOCS:
                return "docs";
            default:
                return "";
        }
    }

    public static File getSourceFileFromUri(Context context, boolean isCompressImage, boolean isTrimVide, BraveFileType braveFileType, String destinationFilePath, Uri contentUri) {
        File file = null;
        String fileType = getFileType(braveFileType);
        if (fileType.equals("image")) {
            //handle image
            String filepath = BraveFileUtils.getFilePathFromContentUri(contentUri, context);
            Bitmap imageBitMap = BraveFileUtils.checkIsBitMapRotated(context, true, new File(filepath));
            if (imageBitMap != null) {
                file = new File(destinationFilePath);
                BraveFileUtils.saveBitmapToExternalStorage(context, imageBitMap, file);
            }
        }
        return file;
    }
}

File: picker/src/main/java/com/braver/tool/picker/BraveFileGetter.kt
/*
 *
 *  * Created by https://github.com/braver-tool on 12/10/21, 08:30 PM
 *  * Copyright (c) 2021 . All rights reserved.
 *  * Last modified 23/03/22, 09:45 AM
 *
 */
package com.braver.tool.picker

import android.content.Context
import android.net.Uri
import java.io.File

object BraveFileGetter {
    private fun getFileType(braveFileType: BraveFileType?): String {
        return when (braveFileType) {
            BraveFileType.IMAGE -> "image"
            BraveFileType.VIDEO -> "video"
            BraveFileType.DOCS -> "docs"
            else -> ""
        }
    }

    @JvmStatic
    fun getSourceFileFromUri(
        context: Context?,
        isCompressImage: Boolean,
        isTrimVide: Boolean,
        braveFileType: BraveFileType?,
        destinationFilePath: String?,
        contentUri: Uri?
    ): File? {
        var file: File? = null
        val fileType = getFileType(braveFileType)
        if (fileType == "image") {
            //handle image
            val filepath = BraveFileUtils.getFilePathFromContentUri(contentUri, context)
            val imageBitMap = BraveFileUtils.checkIsBitMapRotated(context, true, File(filepath))
            if (imageBitMap != null && destinationFilePath != null) {
                file = File(destinationFilePath)
                BraveFileUtils.saveBitmapToExternalStorage(context, imageBitMap, file)
            }
        }
        return file
    }
}